platform/android/dir_access_jandroid.cpp

```diff
@@ -242,8 +242,12 @@ Error DirAccessJAndroid::make_dir_recursive(const String &p_dir) {
 	}
 }
 
-Error DirAccessJAndroid::make_dir(String p_dir) {
-	return make_dir_recursive(p_dir);
+Error DirAccessJAndroid::make_dir_recursive(const String &p_dir) {
+	Error err = make_dir(p_dir);
+	if (err != OK && err != ERR_ALREADY_EXISTS) {
+		ERR_FAIL_V_MSG(err, "Could not create directory: " + p_dir);
+	}
+	return OK;
 }
```

Comment:
Just to be clear, the difference between `make_dir` and `make_dir_recursive` is that `make_dir_recursive` should create all the intermediate directories needed if they're missing. There's also the behavior of it not failing if the directory already exists as done here (like `mkdir -p` on Unix), but it's not clear to me whether the new `make_dir` here also does recursive directory creation.

Reply:
> but it's not clear to me whether the new make_dir here also does recursive directory creation.

Yes it does; on Android both methods point to https://github.com/godotengine/godot/blob/1bd740d18d714f815486b04bf4c6154ef6c355d9/platform/android/java/lib/src/org/godotengine/godot/io/directory/FilesystemDirectoryAccess.kt#L173 which automatically does recursive directory creation. 

The update here is to align the implementation with the other platforms as you described; previously this version of `make_dir_recursive` would return `ERR_ALREADY_EXISTS` if the directory already existed. With this update, it returns `OK` as do the other implementations.